src/interpreter.c

```diff
@@ -384,11 +391,13 @@ static jl_value_t *eval(jl_value_t *e, jl_value_t **locals, size_t nl)
         // temporarily assign so binding is available for field types
         check_can_assign_type(b);
         b->value = (jl_value_t*)dt;
+        gc_wb_binding(b,dt);
```

Comment:
why do you use `gc_wb_binding(((void**)b)-1, dt);` in the other two usages of gc_wb_binding, but not here? @carnaval